### ISSUE TITLE
💅 `noUndeclaredVariables` fails to recognize ambient namespace usage

### ISSUE THREAD

**Issue by jpike88** [biomejs/biome#2487]
### Environment information

```bash
CLI:
  Version:                      1.7.0
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.11.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.2.4"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           true
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### Rule name

`noUndeclaredVariables`

### Playground link

n/a

### Expected result

[biomebug.zip](https://github.com/biomejs/biome/files/15006792/biomebug.zip)

see screenshots

without rule enabled
<img width="553" alt="Screenshot 2024-04-17 at 1 30 51 PM" src="https://github.com/biomejs/biome/assets/9585787/5dc91aa9-543b-4004-b1ca-b2cef9cc7f9a">

with the rule enabled
<img width="776" alt="Screenshot 2024-04-17 at 1 31 18 PM" src="https://github.com/biomejs/biome/assets/9585787/97a122d6-e625-49c0-a32b-f0380fe77c58">

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
We aren't in a position to solve this issue yet because we aren't able to resolve types coming from a third-party library. As for now, I suggest you to disable the rule via overrides or inline suppression
